presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -265,9 +271,21 @@ else if (queryResult.getState() != State.SUCCESS) {
 
     private boolean checkForDeterministicAndRerunTestQueriesIfNeeded()
     {
+        // Decrease hash_partition_count by one if necessary
+        int hashPartitionCount = defaultHashPartitionCount;
+        Map<String, String> sessionPropertiesOverride = null;
+        if (sessionProperties.get(HASH_PARTITION_COUNT) != null) {
+            hashPartitionCount = Integer.parseInt(sessionProperties.get(HASH_PARTITION_COUNT));
+        }
+
+        if (hashPartitionCount > 1) {
+            sessionPropertiesOverride = new HashMap<>(sessionProperties);
```

Comment:
Since we are not going to modify the content of the map, you can use `ImmutableMap` here to enforce it. Check the codebase to see how we construct a immutable map :)